static/app/components/onboarding/gettingStartedDoc/layout.tsx

```diff
@@ -0,0 +1,119 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import HookOrDefault from 'sentry/components/hookOrDefault';
+import ExternalLink from 'sentry/components/links/externalLink';
+import List from 'sentry/components/list';
+import ListItem from 'sentry/components/list/listItem';
+import {Step, StepType} from 'sentry/components/onboarding/gettingStartedDoc/step';
+import {PRODUCT, ProductSelection} from 'sentry/components/onboarding/productSelection';
+import {t} from 'sentry/locale';
+import ConfigStore from 'sentry/stores/configStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import useOrganization from 'sentry/utils/useOrganization';
+
+const ProductSelectionAvailabilityHook = HookOrDefault({
+  hookName: 'component:product-selection-availability',
+});
+
+type NextStep = {
+  description: string;
+  link: string;
+  name: string;
+};
+
+type CurrentSteps = [
+  {
+    code: string;
+    description: React.ReactNode;
+    language: string;
+    type: StepType.INSTALL;
+  },
+  {
+    code: string;
+    description: React.ReactNode;
+    language: string;
+    type: StepType.CONFIGURE;
+  },
+  {
+    code: string;
+    description: React.ReactNode;
+    language: string;
+    type: StepType.VERIFY;
+  }
+];
+
+export type LayoutProps = {
+  steps: CurrentSteps;
+  newOrg?: boolean;
+  nextSteps?: NextStep[];
+};
+
+export function Layout({steps, nextSteps, newOrg}: LayoutProps) {
+  const organization = useOrganization();
+  const {isSelfHosted} = useLegacyStore(ConfigStore);
+
+  return (
+    <Wrapper>
+      {!isSelfHosted && newOrg && (
+        <ProductSelection
+          defaultSelectedProducts={[
+            PRODUCT.PERFORMANCE_MONITORING,
+            PRODUCT.SESSION_REPLAY,
+          ]}
+        />
+      )}
+      {!isSelfHosted && !newOrg && (
+        <ProductSelectionAvailabilityHook organization={organization} />
+      )}
+      <Steps>
+        {steps.map(step => (
+          <Step
+            key={step.type}
+            type={step.type}
+            description={step.description}
+            code={step.code}
+            language={step.language}
+          />
+        ))}
+      </Steps>
+      {nextSteps && (
+        <Fragment>
+          <Divider />
+          <h4>{t('Next Steps')}</h4>
+          <List symbol="bullet">
+            {nextSteps.map(step => (
+              <ListItem key={step.name}>
+                <ExternalLink href={step.link}>{step.name}</ExternalLink>
+                {': '}
+                {step.description}
+              </ListItem>
+            ))}
+          </List>
+        </Fragment>
+      )}
+    </Wrapper>
+  );
+}
+
+const Divider = styled('hr')`
+  height: 1px;
+  width: 100%;
+  background: ${p => p.theme.border};
+  border: none;
+`;
+
+const Steps = styled('div')`
+  display: flex;
+  flex-direction: column;
+  gap: 1.5rem;
```

Comment:
Why rem and not a `space`?

Reply:
was just reusing the styles which are currently being used in the docs. we can polish this afterward